File: packages/lib-model-js/src/ai/LlmPayload/LlmPayload.models.ts
import {
  type LLM_PAYLOAD_TYPE,
  type MESSAGE_ROLE,
} from '@lib/model/ai/LlmPayload/LlmPayload.constants';
import { type DateTime } from '@lib/shared/datetime/utils/DateTime/DateTime';

export type LlmPayloadModel = {
  chat_id: string;
  created: DateTime;
  message_id: string;
  role?: MESSAGE_ROLE;
  text?: string;
  type: LLM_PAYLOAD_TYPE;
};

File: packages/lib-ai-py/src/lib_ai/agent/utils/llm_payload/llm_payload_models.py
# template version: 1.0.0


from datetime import datetime

from lib_shared.core.utils.base_model.base_model import BaseModel
from lib_shared.core.utils.field.field import Field

from lib_ai.agent.utils.ai_message.constants import MessageRole
from lib_ai.agent.utils.llm_payload.constants import LlmPayloadType


class LlmPayloadModel(BaseModel):
    chat_id: str = Field()
    created: datetime = Field(default_value=datetime.now)
    message_id: str = Field()
    role: MessageRole | None = Field(default=None)
    text: str | None = Field(default=None)
    type: LlmPayloadType = Field()
